docs/docs/guides/smart_contracts/index.md

```diff
@@ -1,5 +1,491 @@
 ---
-sidebar_position: 4
+sidebar_position: 2
+sidebar_label: 'Mastering Smart Contracts'
 ---
 
-# Smart Contracts
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Mastering Smart Contracts
+
+:::info
+This guide expects you to have some basic knowledge. If you are just starting, it is recommended to first check out this [Tutorial: Deploying and Interacting with Smart Contracts](./deploying_and_interacting_with_smart_contracts).
+:::
+
+## Contract class
+
+The `Contract` class is the main object exported by the `web3-eth-contract` package. It is also available in the `web3` package.
+
+### Importing the Contract Class 
+
+To use the `Contract` class, you'll need to import it from one of two packages: directly from the `web3-eth-contract` package or from the `web3` package.
+
+Here's an example of importing from each:
+
+<Tabs groupId="prog-lang" queryString>
+  <TabItem value="javascript" label="JavaScript" default 
+  	attributes={{className: "javascript-tab"}}>
+
+```javascript
+// Importing from web3-eth-contract package
+const { Contract } = require("web3-eth-contract");
+const contract = new Contract(...);         
+
+// Importing from the main web3 package
+const { Contract } = require("web3");
+const contract = new Contract(...);
+
+// Importing from the main web3 package from inside `web3.eth` namespace
+const { Web3 } = require("web3");
+const web3 = new Web3("http://127.0.0.1:8545");
+const contract = new web3.eth.Contract(...);
+
+// to set the provider for the contract instance:
+contract.setProvider('http://127.0.0.1:7545');
+```
+
+  </TabItem>
+  <TabItem value="typescript" label="TypeScript"
+  	attributes={{className: "typescript-tab"}}>
+
+
+
+```ts
+// Importing from web3-eth-contract package
+import { Contract } from "web3-eth-contract";
+const contract = new Contract(...);
+
+// Importing from the main web3 package
+import { Contract } from 'web3';
+const contract = new Contract(...);
+
+// Importing from the main web3 package from inside `web3.eth` namespace
+import { Web3 } from 'web3';
+const web3 = new Web3("http://127.0.0.1:8545");
+const contract = new web3.eth.Contract(...);
+
+// to set the provider for the contract instance:
+contract.setProvider('http://127.0.0.1:7545');
+```
+
+  </TabItem>
+</Tabs>
+
+
+### `Contract` vs `web3.eth.Contract`
+
+There is a way to create a contract object, other than the ones listed above. That is by accessing an instance of `Web3` object. And this instance of `Web3` is usually called `web3`.
+
+Actually, the `web3.eth.Contract` is typically how you access the class through a web3 instance that already has a provider setup and that may already has customized configurations.
```

Comment:
I think it would be nice to mention the pro of using the separate web3-eth-contract may be useful if you need to save space on bundle size